docs/testing-guide.md

```diff
@@ -0,0 +1,36 @@
+# Testing
+
+## Prerequisites
+To run tests locally, you can either run `make install-prerequisites` or manually set up the resources below.
+> Note: if you would like to bootstrap a [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) cluster for local testing, run this make target with `TEST_KIND=true`
+
+
+1. A kubernetes cluster
+1. [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+1. [bats](https://github.com/bats-core/bats-core#installing-bats-from-source)
+1. [Helm 3](https://helm.sh/docs/intro/install/)
+
+## Running tests
+To run end to end tests locally, carry out the following steps from root:
+
+1. To bootstrap resources required for testing:
+    ```
+    make e2e-bootstrap
+    ```
+
+1. To deploy the chart with local changes: 
+    ```
+    helm dependency update ./charts/osm-arc
```

Comment:
@ritazh @ksubrmnn I am inclined to include this update command within the `make e2e-helm-deploy` target because it is needed for chart changes every time. Would that be okay or do we want to keep that target as only a helm install?

Reply:
I'd include it in the makefile! That way it'll be harder to miss